graphql/database/userImpact/updateImpact.js

```diff
@@ -0,0 +1,52 @@
+import UserImpactModel from './UserImpactModel'
+import { USER_VISIT_IMPACT_VALUE } from '../constants'
+/**
+ * updates a user impact record
+ * @param {object} userContext - The user authorizer object.
+ * @param {string} userId - The user id.
+ * @param {string} tabId - A UUID for this opened tab
+ * @param {object} updates - options to update impact record, include logImpact, claimPendingReferralImpact, confirmImpact
+ * @return {Promise<User>}  A promise that resolves into a UserImpact instance.
+ */
+
+const updateImpact = async (userContext, userId, charityId, updates) => {
```

Comment:
I see what you're saying about this one mutation doing many things :) It's fine to leave for now, though I agree it's easier to reason about mutations and schema when a mutation is responsible for only one action.